From 3e9321ea1c9f802044f383804987df27123d6a0c Mon Sep 17 00:00:00 2001
From: Michael John <mj@tierone.software>
Date: Sun, 2 Aug 2026 16:08:46 -0400
Subject: [PATCH 06/13] env: check malloc failures in split_spaces

Both allocations were dereferenced unchecked (null deref on OOM,
found by Infer).
---
 src.freebsd/coreutils/env/envopts.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src.freebsd/coreutils/env/envopts.c b/src.freebsd/coreutils/env/envopts.c
index 83e01cc2..ae09b3fe 100644
--- a/src.freebsd/coreutils/env/envopts.c
+++ b/src.freebsd/coreutils/env/envopts.c
@@ -178,6 +178,8 @@ split_spaces(const char *str, int *origind, int *origc, char ***origv)
 	if (*str == '\0')
 		return;
 	newstr = malloc(strlen(str) + 1);
+	if (newstr == NULL)
+		err(1, "malloc");
 
 	/*
 	 * Allocate plenty of space for the new array of arg-pointers,
@@ -185,6 +187,8 @@ split_spaces(const char *str, int *origind, int *origc, char ***origv)
 	 * array.
 	 */
 	newargv = malloc((*origc + (strlen(str) / 2) + 2) * sizeof(char *));
+	if (newargv == NULL)
+		err(1, "malloc");
 	nextarg = newargv;
 	*nextarg++ = **origv;
 
-- 
2.55.0

